Kitodo/src/main/java/org/kitodo/production/interfaces/activemq/TaskActionProcessor.java

```diff
@@ -0,0 +1,212 @@
+            ServiceManager.getProcessService().saveToIndex(currentTask.getProcess(), true);
+
+            for (Task task : currentTask.getProcess().getTasks()) {
+                // update tasks in elastic search index, which includes correction comment status
+                taskService.saveToIndex(task, true);
+            }
+        } catch (DataException | DAOException | CustomResponseException | IOException e) {
+            throw new ProcessorException(e);
+        }
+    }
+
+    private void processAction(MapMessageObjectReader mapMessageObjectReader, TaskAction taskAction, Task currentTask)
+            throws JMSException, ProcessorException, DataException, DAOException, IOException {
+        Comment comment = null;
+        if (mapMessageObjectReader.hasField(KEY_MESSAGE)) {
+            comment = buildComment(currentTask, mapMessageObjectReader.getMandatoryString(KEY_MESSAGE));
+        }
+
+        User currentUser = ServiceManager.getUserService().getCurrentUser();
+        switch (taskAction) {
+            case PROCESS:
+                if (!TaskStatus.OPEN.equals(currentTask.getProcessingStatus())) {
+                    throw new ProcessorException("Status of task is not OPEN.");
+                }
+                actionProcess(currentTask, currentUser);
+                break;
+            case ERROR_OPEN:
+                if (!TaskStatus.OPEN.equals(currentTask.getProcessingStatus()) && !TaskStatus.INWORK.equals(
+                        currentTask.getProcessingStatus())) {
+                    throw new ProcessorException("Status of task is not OPEN or INWORK.");
+                }
+                if (!mapMessageObjectReader.hasField(KEY_MESSAGE)) {
+                    throw new ProcessorException("Message field of task action ERROR_OPEN is required.");
+                }
+                actionErrorOpen(mapMessageObjectReader, currentTask, comment);
+                break;
+            case ERROR_CLOSE:
+                if (!TaskStatus.LOCKED.equals(currentTask.getProcessingStatus())) {
+                    throw new ProcessorException("Status of task is not LOCKED.");
+                }
+                actionErrorClose(mapMessageObjectReader, currentTask, currentUser);
+                break;
+            case CLOSE:
+                workflowControllerService.closeTaskByUser(currentTask);
+                break;
+            default:
+                if (!mapMessageObjectReader.hasField(KEY_MESSAGE)) {
+                    throw new ProcessorException("Message field of task action COMMENT is required.");
+                }
+        }
+
+        if (Objects.nonNull(comment)) {
+            ServiceManager.getCommentService().saveToDatabase(comment);
+        }
+    }
+
+    private static Comment buildComment(Task currentTask, String message) {
+        Comment comment = new Comment();
+        comment.setProcess(currentTask.getProcess());
+        comment.setAuthor(ServiceManager.getUserService().getCurrentUser());
```

Comment:
Does this work? Does Active MQ have a “current user”? (Just asking. Not necessary to change if it is okay.)